src/utils/auth.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable import/prefer-default-export */
+
+import createAuth0Client from '@auth0/auth0-spa-js';
+
+let auth0 = null;
+
+function fetchAuthConfig() {
+  return fetch('__BASE_DOMAIN__/api/v1/front-end-auth-config/');
+}
+
+async function configureClient() {
+  const response = await fetchAuthConfig();
+  const config = await response.json();
+
+  auth0 = await createAuth0Client({
+    domain: config.domain,
+    client_id: config.clientID,
+    audience: config.audience,
+  });
+}
+
+async function withToken(cb) {
+  cb(await auth0.getTokenSilently());
+}
+
+export function authenticate(successCallback) {
+  window.onload = async () => {
+    await configureClient();
+
+    const isAuthenticated = await auth0.isAuthenticated();
+
+    if (isAuthenticated) {
+      withToken(successCallback);
+    } else {
+      const query = window.location.search;
+      if (query.includes('code=') && query.includes('state=')) {
+        await auth0.handleRedirectCallback();
+        window.history.replaceState({}, document.title, '/');
```

Comment:
help me understand – will this remove the query string if it existed before? At the moment, GLAM & GUD work similarly in that we are capturing the relevant parts of the app state in the url. It looks like the redirect done this way might clobber that.

Reply:
Ah, yes. I took this from the tutorial thinking that it'd be fine (it'll only happen when the user first visits the site), but this will also break links and bookmarks. Good call. Thank you!